tools-v2/pkg/cli/command/curvebs/update/leader/leader.go

```diff
@@ -0,0 +1,207 @@
+	Info    *basecmd.Rpc
+	Request *cli2.TransferLeaderRequest2
+	Client  cli2.CliService2Client
+}
+
+func (lRpc *LeaderRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	lRpc.Client = cli2.NewCliService2Client(cc)
+}
+
+func (lRpc *LeaderRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return lRpc.Client.TransferLeader(ctx, lRpc.Request)
+}
+
+type leaderCommand struct {
+	basecmd.FinalCurveCmd
+
+	// request parameters
+	opts          peer.Options
+	logicalPoolID uint32
+	copysetID     uint32
+
+	conf       peer.Configuration
+	leaderPeer *common.Peer
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*leaderCommand)(nil) // check interface
+
+// NewCommand ...
+func NewleaderCommand() *cobra.Command {
+	peerCmd := &leaderCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "leader",
+			Short:   "update(reset) the leader from the copyset",
+			Example: leaderExample,
+		},
+	}
+	basecmd.NewFinalCurveCli(&peerCmd.FinalCurveCmd, peerCmd)
+	return peerCmd.Cmd
+}
+
+func (lCmd *leaderCommand) AddFlags() {
+	config.AddRpcRetryTimesFlag(lCmd.Cmd)
+	config.AddRpcTimeoutFlag(lCmd.Cmd)
+
+	config.AddBSLogicalPoolIdRequiredFlag(lCmd.Cmd)
+	config.AddBSCopysetIdRequiredFlag(lCmd.Cmd)
+
+	config.AddBSPeersConfFlag(lCmd.Cmd)
+}
+
+func (lCmd *leaderCommand) Init(cmd *cobra.Command, args []string) error {
+	lCmd.opts = peer.Options{}
+
+	lCmd.SetHeader([]string{cobrautil.ROW_LEADER, cobrautil.ROW_OLDLEADER, cobrautil.ROW_COPYSET, cobrautil.ROW_RESULT, cobrautil.ROW_REASON})
+	lCmd.TableNew.SetAutoMergeCellsByColumnIndex(cobrautil.GetIndexSlice(
+		lCmd.Header, []string{},
+	))
+
+	lCmd.opts.Timeout = config.GetFlagDuration(lCmd.Cmd, config.RPCTIMEOUT)
+	lCmd.opts.RetryTimes = config.GetFlagInt32(lCmd.Cmd, config.RPCRETRYTIMES)
+
+	lCmd.copysetID = config.GetBsFlagUint32(lCmd.Cmd, config.CURVEBS_COPYSET_ID)
+	lCmd.logicalPoolID = config.GetBsFlagUint32(lCmd.Cmd, config.CURVEBS_LOGIC_POOL_ID)
+
+	// parse peers config
+	peers := config.GetBsFlagStringSlice(lCmd.Cmd, config.CURVEBS_PEERS_ADDRESS)
+	c, e := peer.ParseConfiguration(peers)
+	if e != nil {
+		return e.ToError()
+	}
+	lCmd.conf = *c
+
+	// parse peer conf
+	if len(args) < 1 {
+		pErr := cmderror.ErrGetPeer()
+		pErr.Format("should specified the peer address")
+		return pErr.ToError()
+	}
+	peerPara := args[0]
+	lCmd.leaderPeer, e = peer.ParsePeer(peerPara)
+	if e != nil {
+		return e.ToError()
+	}
+	return nil
+}
+
+func (lCmd *leaderCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&lCmd.FinalCurveCmd, lCmd)
+}
+
+func (lCmd *leaderCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	out := make(map[string]string)
+	out[cobrautil.ROW_LEADER] = fmt.Sprintf("%s:%d", lCmd.leaderPeer.GetAddress(), lCmd.leaderPeer.GetId())
+	out[cobrautil.ROW_COPYSET] = fmt.Sprintf("(%d:%d)", lCmd.logicalPoolID, lCmd.copysetID)
+
+	// 1. acquire leader peer info.
+	leader, err := peer.GetLeader(lCmd.logicalPoolID, lCmd.copysetID, lCmd.conf, lCmd.opts)
```

Comment:
Wouldn't it be better to define this as a command?

Reply:
will do it in next pr.